main.py

```diff
@@ -104,114 +126,164 @@
-- When users ask for Chronicle/SIEM queries, threat intelligence, SOAR cases, or SCC findings, handle directly with your MCP tools
-- After delegation, review the sub-agent's response and provide additional context or follow-up actions using your security tools if needed
+YOUR CAPABILITIES:
+- Retrieve security Runbooks, IRPs, Common Steps, Procedures, guidelines, and personas using retrieve_agentic_soc_runbooks tool
+- Query Chronicle/SIEM for security events and detections
+- Manage SOAR cases and incidents
+- Access threat intelligence through GTI tools
+- Retrieve SCC findings and cloud security posture
+- Use list_tools to get full MCP Tool list
 
-COMMUNICATION PATTERN:
-1. If the request involves searching runbooks/procedures: "I'll search our runbooks for that information" → delegate to vertex_search_agent
-2. Review sub-agent results and enhance with relevant security tool actions
-3. Provide comprehensive response combining runbook guidance with live security data
+WORKFLOW:
+1. When users ask about runbooks or procedures, use the retrieve_agentic_soc_runbooks tool to retrieve relevant documentation from the RAG corpus
+2. For security investigations, combine runbook guidance with live data from Chronicle, GTI, and SOAR
+3. Provide comprehensive responses that integrate procedural knowledge with real-time security data
 
-Use your MCP tools for real-time security operations and delegate documentation searches to ensure complete responses.""",
+KEY TOOLS:
+- retrieve_agentic_soc_runbooks: Retrieve security procedures and documentation from the RAG corpus
+- Chronicle tools: Query SIEM for security events
+- SOAR tools: Manage cases and incidents
+- GTI tools: Get threat intelligence
+- SCC tools: Cloud security findings
+
+Always provide actionable guidance combining documented procedures with live security data.""",
   tools=[
      secops_soar_tools,
      secops_siem_tools,
      gti_tools,
      scc_tools,
-  ],
-  sub_agents=[vertex_search_agent]
+     # vertex_search_tools,
+     ask_vertex_retrieval,
+  ]
 )
 
 vertexai.init(
-    project=PROJECT_ID,
-    location=LOCATION,
-    staging_bucket=STAGING_BUCKET,
+    project=GCP_PROJECT_ID,
+    location=GCP_LOCATION,
+    staging_bucket=GCP_STAGING_BUCKET,
 )
 
-# copy the JSON SA file to the same dir as server.py
-shutil.copy(SECOPS_SA_PATH, "./mcp-security/server/secops/secops_mcp/")
+# copy the JSON SA file to the same dir as server.py; ToDo: move to Secret Mgr
+shutil.copy(CHRONICLE_SERVICE_ACCOUNT_PATH, "./mcp-security/server/secops/secops_mcp/")
+# puts file into agent_engine_dependencies.tar.gz to record state of build (not used as code)
+shutil.copy("main.py", "./mcp-security/server/main.txt")
+# shutil.copy(SECOPS_SA_PATH, "./mcp-security/server/vertex-search/")  # disabled while researching
+
+def session_service_builder():
+  from google.adk.sessions import InMemorySessionService
+  return InMemorySessionService()
```

Comment:
This only works about 20% of the time but the other services provided by `google.adk.sessions` (VertexAiSessionService, DatabaseSessionService, etc.) work 0% of the time, so it is the only option.

Note that VertexAiSessionService also requires `GOOGLE_CLOUD_AGENT_ENGINE_ID` and I don't understand how that would be available since thtat is the thing we are building here. Also, that env var is used as a param to agent_engine_id VertexAiSessionService and seems to be added in google-adk==1.15.x., which doesn't work under any combination of these settings.

Use git blame on these files to research: https://github.com/search?q=repo%3Agoogleapis%2Fpython-aiplatform%20GOOGLE_CLOUD_AGENT_ENGINE_ID&type=code

Reply:
Fixed! The google-cloud-aiplatform released today fixes it: 
 * Upgrade google-adk from 1.14.1 to 1.15.1 to fix 'static_instruction' AttributeError
 * Pin google-cloud-aiplatform[agent-engines] to 1.119.0 for VertexAiSessionService compatibility